napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1607 @@
+        the initial coordinates of the drag. `None` otherwise.
+    _drag_box : None | np.ndarray
+        If a drag box is being created to select shapes then this is a 2x2
+        array of the two extreme corners of the drag. `None` otherwise.
+    _drag_box_stored : None | np.ndarray
+        If a drag box is being created to select shapes then this is a 2x2
+        array of the two extreme corners of the drag that have previously been
+        rendered. `None` otherwise. Used to prevent rerendering the same
+        drag box when no data has changed.
+    _is_moving : bool
+        Bool indicating if any shapes are currently being moved.
+    _is_selecting : bool
+        Bool indicating if a drag box is currently being created in order to
+        select shapes.
+    _is_creating : bool
+        Bool indicating if any shapes are currently being created.
+    _fixed_aspect : bool
+        Bool indicating if aspect ratio of shapes should be preserved on
+        resizing.
+    _aspect_ratio : float
+        Value of aspect ratio to be preserved if `_fixed_aspect` is `True`.
+    _fixed_vertex : None | np.ndarray
+        If a scaling or rotation is in progress then a length 2 array of the
+        coordinates that are remaining fixed during the move. `None` otherwise.
+    _fixed_index : int
+        If a scaling or rotation is in progress then the index of the vertex of
+        the boudning box that is remaining fixed during the move. `None`
+        otherwise.
+    _cursor_coord : np.ndarray
+        Length 2 array of the current cursor position in Image coordinates.
+    _update_properties : bool
+        Bool indicating if properties are to allowed to update the selected
+        shapes when they are changed. Blocking this prevents circular loops
+        when shapes are selected and the properties are changed based on that
+        selection
+    _clipboard : list
+        List of shape objects that are to be used during a copy and paste.
+    _colors : list
+        List of supported vispy color names.
+    _vertex_size : float
+        Size of the vertices of the shapes and boudning box in Canvas
+        coordinates.
+    _rotation_handle_length : float
+        Length of the rotation handle of the boudning box in Canvas
+        coordinates.
+    _highlight_color : list
+        Length 3 list of color used to highlight shapes and the interaction
+        box.
+    _highlight_width : float
+        Width of the edges used to highlight shapes.
+    """
+
+    _colors = get_color_names()
+    _vertex_size = 10
+    _rotation_handle_length = 20
+    _highlight_color = (0, 0.6, 1)
+    _highlight_width = 1.5
+
+    def __init__(self, data, *, shape_type='rectangle', edge_width=1,
+                 edge_color='black', face_color='white', opacity=0.7,
+                 z_index=0, name=None):
+
+        visual = Compound([Markers(), VispyLine(), Mesh(), Mesh()])
```

Comment:
Can you make a comment that explains what these things are?

Reply:
done